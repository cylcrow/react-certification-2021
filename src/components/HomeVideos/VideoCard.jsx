import React from 'react';
import { 
    Card, 
    CardImage, 
    CardTitle, 
    CardContent, 
    CardDescription 
} from "../ui/index"

const VideoCard = ({ video }) => {

    const image = video.snippet.thumbnails.high.url;
    const { title, description } = video.snippet;

    return (<Card role="figure">
        <CardImage src={image} alt={title}/>
        <CardContent>
            <CardTitle title="video-title">{ title }</CardTitle>
            <CardDescription >{ description }</CardDescription>
        </CardContent>
    </Card>);
};

export default VideoCard;