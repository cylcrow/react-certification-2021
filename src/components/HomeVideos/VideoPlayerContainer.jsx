import React, { useEffect } from 'react';
import styled from 'styled-components';

const StyledVideoPlayerContainer = styled.div``;

const VideoPlayerContainer = () => {
  useEffect(() => {
    let done = false;
    /* global YT */
    /* eslint no-undef: "error" */
    const player = new YT.Player('player', {
      height: '390',
      width: '640',
      videoId: 'M7lc1UVf-VE',
      events: {
        onReady(event) {
          event.target.playVideo();
        },
        onStateChange(event) {
          if (event.data === YT.PlayerState.PLAYING && !done) {
            setTimeout(player.stopVideo, 6000);
            done = true;
          }
        },
      },
    });
  });

  return (
    <StyledVideoPlayerContainer>
      <div id="player" />
    </StyledVideoPlayerContainer>
  );
};

export default VideoPlayerContainer;
