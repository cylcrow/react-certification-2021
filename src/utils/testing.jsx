import React from "react";
import { render } from '@testing-library/react';
import { ThemeProvider } from "styled-components";
import { lightTheme } from '../providers/theme/themes';

export const renderWithTheme = (Component, theme = lightTheme) => {
    return render(
        <ThemeProvider theme={{ theme, switchTheme: jest.fn() }}>
        { Component }
        </ThemeProvider>
    );
}