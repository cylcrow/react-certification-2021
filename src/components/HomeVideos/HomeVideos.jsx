import React from 'react';
import styled from 'styled-components';
import { data } from './mockData';
import VideoCard from './VideoCard';

const StyledDiv = styled.div`
  height: 100%;
  width: 100%;
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
`;

const HomeVideos = () => {
  return (
    <StyledDiv>
      {data.items.map((video) => {
        const key = JSON.stringify(video.id);
        return (
          <div key={key} data-testid={`video-card-${key}`}>
            <VideoCard video={video} />
          </div>
        );
      })}
    </StyledDiv>
  );
};

export default HomeVideos;
