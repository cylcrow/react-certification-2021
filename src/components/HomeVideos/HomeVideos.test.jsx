import React from 'react';
import '../../utils/testing';
import 'jest-styled-components';
import { render, unmountComponentAtNode } from 'react-dom';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import HomeVideos from './HomeVideos';
import { getAllByTestId } from '@testing-library/react';
import { data } from "./mockData"

let container;
const build = () => {
  render(<HomeVideos />, container);
};

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
  return container;
});

describe('HomeVideos', () => {
  it('renders', () => {
    const wrapper = build(<HomeVideos />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('HomeVideos styles and props', () => {
  it('applies default styling', () => {
    const tree = mount(<HomeVideos />);
    expect(tree).toHaveStyleRule('height', '100%');
    expect(tree).toHaveStyleRule('width', '100%');
  });
});

describe('HomeVideos shows home videos', () => {
  it('displays all videos', () => {
    build();
    const videos = getAllByTestId(container, (id) => id.includes("video-card-"));
    expect(videos).toHaveLength(data.items.length);
  })
})
