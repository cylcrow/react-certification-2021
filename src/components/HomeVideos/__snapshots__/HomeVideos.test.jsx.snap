// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeVideos renders 1`] = `undefined`;
